### DISCUSSION TITLE
Change config from cli

### DISCUSSION THREAD

**Discussion by TheRaven500** [Motion-Project/motionplus#171] [Q&A] [answered]
Is there an option in motionplus to change config from cli? In motion i can control the settings from cli like this:
`curl --silent --max-time 2 --connect-timeout 2 http://localhost:8080/0/detection/status`
Or change it like that:
`curl --silent --max-time 2 --connect-timeout 2 http://localhost:8080/0/detection/pause`
Same for the settings:
`curl --silent --max-time 2 --connect-timeout 2 http://localhost:8080/0/config/set?vid_control_params="brightness=60,contrast=40"`

**Comment by Mr-Dave** [comment, marked as answer]
There is.....it is just that the documentation doesn't exist on how....

Status is via a json page.  e.g.

`http://localhost:8080/0/status.json`

Others are now "post" actions in Motionplus versus them being "get" actions in Motion.  Below are some examples.  (camid would vary by user.  Use camid=0 for all cameras)

Available commands:
`eventstart, eventend, snapshot, pause, unpause, restart, stop, config_write, camera_add, camera_delete, config, action_user, pan_left, pan_right, tilt_up, tilt_down, zoom_in, zoom_out`

Usage:
`curl -s -o -X POST http://localhost:8080/ --data "camid=101" --data "command=eventend"  1>/dev/null`
`curl -s -o -X POST http://localhost:8080/ --data "camid=101" --data "command=config" --data "framerate=12"  1>/dev/null`

I'd note that setting of the configuration parameters via the web interface needs additional work and structural changes to the code (e.g.  The class conversion process currently being implemented).   As a result, you'll probably need to restart any camera affected by a configuration revision to see the change.

**Reply by TheRaven500** [reply]
Thank you very much for answering (and also for fixing the hostname bug). I will try it ASAP. My project (codename "WetBox) has a PiR sensor which triggers the event. So i need to start/end the event via command. And i also need to change config for nightvision (more contrast and more brightness).
